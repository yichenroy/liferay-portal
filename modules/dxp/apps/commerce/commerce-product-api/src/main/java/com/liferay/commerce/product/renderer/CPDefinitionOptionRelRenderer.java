/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 * <p>
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * <p>
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.renderer;

import aQute.bnd.annotation.ProviderType;

import com.liferay.portal.kernel.exception.PortalException;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 * @author Marco Leo
 */
@ProviderType
public interface CPDefinitionOptionRelRenderer {

	public String render(
		long cpDefinitionId, HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse) throws PortalException;

}