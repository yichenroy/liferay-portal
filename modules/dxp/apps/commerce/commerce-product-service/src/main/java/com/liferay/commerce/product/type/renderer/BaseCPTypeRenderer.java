/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.type.renderer;

import com.liferay.commerce.product.type.CPTypeRenderer;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.PortletConstants;
import com.liferay.portal.kernel.portlet.PortletBag;
import com.liferay.portal.kernel.portlet.PortletBagPool;
import com.liferay.portal.kernel.util.Portal;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;

import java.io.IOException;

import javax.servlet.RequestDispatcher;
import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
public abstract class BaseCPTypeRenderer implements CPTypeRenderer {

	public void renderJSP(
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		renderJSP(
			getServletContext(httpServletRequest), httpServletRequest,
			httpServletResponse, path);
	}

	public void renderJSP(
			ServletContext servletContext,
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		RequestDispatcher requestDispatcher =
			servletContext.getRequestDispatcher(path);

		try {
			requestDispatcher.include(httpServletRequest, httpServletResponse);
		}
		catch (ServletException se) {
			_log.error("Unable to render JSP " + path, se);

			throw new IOException("Unable to render " + path, se);
		}
	}

	public void setServletContext(ServletContext servletContext) {
		_servletContext = servletContext;
	}

	protected ServletContext getServletContext(
		HttpServletRequest httpServletRequest) {

		if (_servletContext != null) {
			return _servletContext;
		}

		String portletId = portal.getPortletId(httpServletRequest);

		if (Validator.isNotNull(portletId)) {
			String rootPortletId = PortletConstants.getRootPortletId(portletId);

			PortletBag portletBag = PortletBagPool.get(rootPortletId);

			return portletBag.getServletContext();
		}

		return (ServletContext)httpServletRequest.getAttribute(WebKeys.CTX);
	}

	@Reference
	protected Portal portal;

	private static final Log _log = LogFactoryUtil.getLog(
		BaseCPTypeRenderer.class);

	private ServletContext _servletContext;

}