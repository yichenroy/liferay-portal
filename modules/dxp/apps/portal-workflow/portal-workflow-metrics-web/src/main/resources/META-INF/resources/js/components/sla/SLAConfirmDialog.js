import React from 'react';

export default class SLAConfirmDialog extends React.Component {
	render() {
		const {item: itemRemoved} = this.props;

		return (
			<div className="modal" role="dialog" tabIndex="-1">
				<div className="modal-dialog modal-lg">
					<div className="modal-content">
						<div className="modal-body">
							<p>
								{Liferay.Language.get(
									'deleting-slas-will-reflect-on-report-data'
								)}
							</p>
						</div>

						<div className="modal-footer">
							<div className="modal-item-last">
								<div className="btn-group">
									<div className="btn-group-item">
										<button className="btn btn-secondary" type="button">
											{Liferay.Language.get('cancel')}
										</button>
									</div>

									<div className="btn-group-item">
										<button className="btn btn-secondary" type="button">
											{Liferay.Language.get('ok')}
										</button>
									</div>
								</div>
							</div>
						</div>
					</div>
				</div>
			</div>
		);
	}
}