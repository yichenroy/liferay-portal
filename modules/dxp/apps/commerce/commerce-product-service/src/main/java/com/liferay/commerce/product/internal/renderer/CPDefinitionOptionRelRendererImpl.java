/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.internal.renderer;

import com.liferay.commerce.product.model.CPDefinitionOptionRel;
import com.liferay.commerce.product.model.CPDefinitionOptionValueRel;
import com.liferay.commerce.product.renderer.CPDefinitionOptionRelRenderer;
import com.liferay.commerce.product.service.CPDefinitionOptionRelService;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderer;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderingContext;
import com.liferay.dynamic.data.mapping.model.DDMForm;
import com.liferay.dynamic.data.mapping.model.DDMFormField;
import com.liferay.dynamic.data.mapping.model.DDMFormFieldOptions;
import com.liferay.dynamic.data.mapping.model.LocalizedValue;
import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.util.Portal;

import java.util.List;
import java.util.Locale;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
@Component(immediate = true)
public class CPDefinitionOptionRelRendererImpl
	implements CPDefinitionOptionRelRenderer {

	@Override
	public String render(
			long cpDefinitionId, HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse)
		throws PortalException {

		DDMForm ddmForm = new DDMForm();

		Locale locale = _portal.getLocale(httpServletRequest);

		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
			_cpDefinitionOptionRelService.
				getSkuContributorCPDefinitionOptionRels(cpDefinitionId);

		for (CPDefinitionOptionRel cpDefinitionOptionRel :
				cpDefinitionOptionRels) {

			List<CPDefinitionOptionValueRel> cpDefinitionOptionValueRels =
				cpDefinitionOptionRel.getCPDefinitionOptionValueRels();

			DDMFormField ddmFormField = new DDMFormField(
				String.valueOf(cpDefinitionOptionRel.
					getCPDefinitionOptionRelId()),
				cpDefinitionOptionRel.getDDMFormFieldTypeName());

			if (!cpDefinitionOptionValueRels.isEmpty()) {
				DDMFormFieldOptions ddmFormFieldOptions =
					new DDMFormFieldOptions();

				for (CPDefinitionOptionValueRel cpDefinitionOptionValueRel :
						cpDefinitionOptionValueRels) {

					ddmFormFieldOptions.addOptionLabel(
						String.valueOf(cpDefinitionOptionValueRel.
							getCPDefinitionOptionValueRelId()),
						locale, cpDefinitionOptionValueRel.getTitle(locale));
				}

				ddmFormField.setDDMFormFieldOptions(ddmFormFieldOptions);
			}

			LocalizedValue localizedValue = new LocalizedValue(locale);

			localizedValue.addString(
				locale, cpDefinitionOptionRel.getName(locale));

			ddmFormField.setLabel(localizedValue);

			ddmFormField.setRequired(true);

			ddmForm.addDDMFormField(ddmFormField);
		}

		DDMFormRenderingContext ddmFormRenderingContext =
			new DDMFormRenderingContext();

		ddmFormRenderingContext.setContainerId("options");
		ddmFormRenderingContext.setHttpServletRequest(httpServletRequest);
		ddmFormRenderingContext.setHttpServletResponse(httpServletResponse);
		ddmFormRenderingContext.setLocale(locale);

		return _ddmFormRenderer.render(ddmForm, ddmFormRenderingContext);
	}

	@Reference
	private CPDefinitionOptionRelService _cpDefinitionOptionRelService;

	@Reference
	private DDMFormRenderer _ddmFormRenderer;

	@Reference
	private Portal _portal;

}