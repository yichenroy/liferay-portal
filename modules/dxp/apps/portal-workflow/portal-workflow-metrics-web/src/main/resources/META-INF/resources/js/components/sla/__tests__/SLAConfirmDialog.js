import React from 'react';
import renderer from 'react-test-renderer';
import SLAConfirmDialog from '../SLAConfirmDialog';

test('Should render component', () => {
	const component = renderer.create(<SLAConfirmDialog item={'test'} />);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});