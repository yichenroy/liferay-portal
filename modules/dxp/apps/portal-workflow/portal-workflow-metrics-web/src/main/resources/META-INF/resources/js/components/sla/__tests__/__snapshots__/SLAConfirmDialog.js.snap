// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should cancel dialog 1`] = `
<MockRouter>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/slas/1234",
          "search": "?backPath=%2Fprocesses",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/slas/1234",
              "search": "?backPath=%2Fprocesses",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 3,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SLAConfirmDialog
        itemToRemove={1234}
      >
        <div
          className="modal"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog modal-lg"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-body"
              >
                <p>
                  Deleting SLAs will reflect on report data. This action cannot be undone. Are you sure you want to permanently delete it?
                </p>
              </div>
              <div
                className="modal-footer"
              >
                <div
                  className="modal-item-last"
                >
                  <div
                    className="btn-group"
                  >
                    <div
                      className="btn-group-item"
                    >
                      <button
                        className="btn btn-secondary"
                        onClick={[Function]}
                        type="button"
                      >
                        Cancel
                      </button>
                    </div>
                    <div
                      className="btn-group-item"
                    >
                      <button
                        className="btn btn-secondary"
                        id="remove_sla_button"
                        onClick={[Function]}
                        type="button"
                      >
                        ok
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </SLAConfirmDialog>
    </Router>
  </MemoryRouter>
</MockRouter>
`;

exports[`Should cancel dialog through SLA List 1`] = `
<MockRouter
  client={
    Object {
      "delete": [Function],
      "get": [Function],
      "post": [Function],
      "put": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/slas/1234",
          "search": "?backPath=%2Fprocesses",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/slas/1234",
              "search": "?backPath=%2Fprocesses",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 3,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SLAListCard>
        <nav
          className="management-bar management-bar-light navbar navbar-expand-md"
        >
          <div
            className="container-fluid container-fluid-max-xl"
          >
            <ul
              className="navbar-nav autofit-row"
            >
              <li
                className="nav-item autofit-col-expand autofit-float-end"
              >
                <withRouter()
                  className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                  to="/sla/new/undefined"
                >
                  <Route>
                    <Component
                      className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "canGo": [Function],
                          "createHref": [Function],
                          "entries": Array [
                            Object {
                              "hash": "",
                              "match": Object {
                                "params": Object {
                                  "page": 1,
                                  "search": undefined,
                                  "sort": undefined,
                                },
                              },
                              "pathname": "/processes/1/10/title%3Aasc",
                              "search": "?backPath=%2F",
                            },
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "?backPath=%2F",
                            },
                            Object {
                              "hash": "",
                              "pathname": "/slas/1234",
                              "search": "?backPath=%2Fprocesses",
                            },
                          ],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "index": 0,
                          "length": 3,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "match": Object {
                              "params": Object {
                                "page": 1,
                                "search": undefined,
                                "sort": undefined,
                              },
                            },
                            "pathname": "/processes/1/10/title%3Aasc",
                            "search": "?backPath=%2F",
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        }
                      }
                      match={
                        Object {
                          "isExact": false,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      to="/sla/new/undefined"
                    >
                      <ChildLinkWrapper
                        backPath="/"
                        className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                        currentPath="/processes/1/10/title%3Aasc?backPath=%2F"
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "canGo": [Function],
                            "createHref": [Function],
                            "entries": Array [
                              Object {
                                "hash": "",
                                "match": Object {
                                  "params": Object {
                                    "page": 1,
                                    "search": undefined,
                                    "sort": undefined,
                                  },
                                },
                                "pathname": "/processes/1/10/title%3Aasc",
                                "search": "?backPath=%2F",
                              },
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "?backPath=%2F",
                              },
                              Object {
                                "hash": "",
                                "pathname": "/slas/1234",
                                "search": "?backPath=%2Fprocesses",
                              },
                            ],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "index": 0,
                            "length": 3,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "match": Object {
                                "params": Object {
                                  "page": 1,
                                  "search": undefined,
                                  "sort": undefined,
                                },
                              },
                              "pathname": "/processes/1/10/title%3Aasc",
                              "search": "?backPath=%2F",
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        to="/sla/new/undefined"
                      >
                        <Link
                          className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                          replace={false}
                          to={
                            Object {
                              "pathname": "/sla/new/undefined",
                              "search": "?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F",
                            }
                          }
                        >
                          <a
                            className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                            href="/sla/new/undefined?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
                            onClick={[Function]}
                          >
                            <Icon
                              iconName="plus"
                            >
                              <svg
                                className="lexicon-icon lexicon-icon-plus"
                                focusable="false"
                                role="presentation"
                              >
                                <use
                                  href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#plus"
                                />
                              </svg>
                            </Icon>
                          </a>
                        </Link>
                      </ChildLinkWrapper>
                    </Component>
                  </Route>
                </withRouter()>
              </li>
            </ul>
          </div>
        </nav>
        <div
          className="lfr-search-container-wrapper container-fluid-1280"
        >
          <ListView
            emptyMessageText="SLA allows to define and measure process performance."
            emptyTitleText="No SLAs Yet"
            isFetching={false}
            isLoading={true}
          >
            <div>
              <LoadingState>
                <span
                  aria-hidden="true"
                  className="loading-animation"
                />
              </LoadingState>
            </div>
          </ListView>
        </div>
      </SLAListCard>
    </Router>
  </MemoryRouter>
</MockRouter>
`;

exports[`Should cancel dialog through SLA List 2`] = `
<MockRouter
  client={
    Object {
      "delete": [Function],
      "get": [Function],
      "post": [Function],
      "put": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/slas/1234",
          "search": "?backPath=%2Fprocesses",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/slas/1234",
              "search": "?backPath=%2Fprocesses",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 3,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SLAListCard>
        <nav
          className="management-bar management-bar-light navbar navbar-expand-md"
        >
          <div
            className="container-fluid container-fluid-max-xl"
          >
            <ul
              className="navbar-nav autofit-row"
            >
              <li
                className="nav-item autofit-col-expand autofit-float-end"
              >
                <withRouter()
                  className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                  to="/sla/new/undefined"
                >
                  <Route>
                    <Component
                      className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "canGo": [Function],
                          "createHref": [Function],
                          "entries": Array [
                            Object {
                              "hash": "",
                              "match": Object {
                                "params": Object {
                                  "page": 1,
                                  "search": undefined,
                                  "sort": undefined,
                                },
                              },
                              "pathname": "/processes/1/10/title%3Aasc",
                              "search": "?backPath=%2F",
                            },
                            Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "?backPath=%2F",
                            },
                            Object {
                              "hash": "",
                              "pathname": "/slas/1234",
                              "search": "?backPath=%2Fprocesses",
                            },
                          ],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "index": 0,
                          "length": 3,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "match": Object {
                              "params": Object {
                                "page": 1,
                                "search": undefined,
                                "sort": undefined,
                              },
                            },
                            "pathname": "/processes/1/10/title%3Aasc",
                            "search": "?backPath=%2F",
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        }
                      }
                      match={
                        Object {
                          "isExact": false,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      to="/sla/new/undefined"
                    >
                      <ChildLinkWrapper
                        backPath="/"
                        className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                        currentPath="/processes/1/10/title%3Aasc?backPath=%2F"
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "canGo": [Function],
                            "createHref": [Function],
                            "entries": Array [
                              Object {
                                "hash": "",
                                "match": Object {
                                  "params": Object {
                                    "page": 1,
                                    "search": undefined,
                                    "sort": undefined,
                                  },
                                },
                                "pathname": "/processes/1/10/title%3Aasc",
                                "search": "?backPath=%2F",
                              },
                              Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "?backPath=%2F",
                              },
                              Object {
                                "hash": "",
                                "pathname": "/slas/1234",
                                "search": "?backPath=%2Fprocesses",
                              },
                            ],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "index": 0,
                            "length": 3,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "match": Object {
                                "params": Object {
                                  "page": 1,
                                  "search": undefined,
                                  "sort": undefined,
                                },
                              },
                              "pathname": "/processes/1/10/title%3Aasc",
                              "search": "?backPath=%2F",
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        to="/sla/new/undefined"
                      >
                        <Link
                          className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                          replace={false}
                          to={
                            Object {
                              "pathname": "/sla/new/undefined",
                              "search": "?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F",
                            }
                          }
                        >
                          <a
                            className="btn btn-primary nav-btn nav-btn-monospaced navbar-breakpoint-down-d-none"
                            href="/sla/new/undefined?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
                            onClick={[Function]}
                          >
                            <Icon
                              iconName="plus"
                            >
                              <svg
                                className="lexicon-icon lexicon-icon-plus"
                                focusable="false"
                                role="presentation"
                              >
                                <use
                                  href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#plus"
                                />
                              </svg>
                            </Icon>
                          </a>
                        </Link>
                      </ChildLinkWrapper>
                    </Component>
                  </Route>
                </withRouter()>
              </li>
            </ul>
          </div>
        </nav>
        <div
          className="lfr-search-container-wrapper container-fluid-1280"
        >
          <ListView
            emptyMessageText="SLA allows to define and measure process performance."
            emptyTitleText="No SLAs Yet"
            isFetching={false}
            isLoading={true}
          >
            <div>
              <LoadingState>
                <span
                  aria-hidden="true"
                  className="loading-animation"
                />
              </LoadingState>
            </div>
          </ListView>
        </div>
      </SLAListCard>
    </Router>
  </MemoryRouter>
</MockRouter>
`;

exports[`Should remove item 1`] = `
<MockRouter>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/slas/1234",
          "search": "?backPath=%2Fprocesses",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/slas/1234",
              "search": "?backPath=%2Fprocesses",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 3,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SLAConfirmDialog
        itemToRemove={1234}
      >
        <div
          className="modal"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog modal-lg"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-body"
              >
                <p>
                  Deleting SLAs will reflect on report data. This action cannot be undone. Are you sure you want to permanently delete it?
                </p>
              </div>
              <div
                className="modal-footer"
              >
                <div
                  className="modal-item-last"
                >
                  <div
                    className="btn-group"
                  >
                    <div
                      className="btn-group-item"
                    >
                      <button
                        className="btn btn-secondary"
                        onClick={[Function]}
                        type="button"
                      >
                        Cancel
                      </button>
                    </div>
                    <div
                      className="btn-group-item"
                    >
                      <button
                        className="btn btn-secondary"
                        id="remove_sla_button"
                        onClick={[Function]}
                        type="button"
                      >
                        ok
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </SLAConfirmDialog>
    </Router>
  </MemoryRouter>
</MockRouter>
`;

exports[`Should render component 1`] = `
<div
  className="modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-lg"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-body"
      >
        <p>
          Deleting SLAs will reflect on report data. This action cannot be undone. Are you sure you want to permanently delete it?
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <div
          className="modal-item-last"
        >
          <div
            className="btn-group"
          >
            <div
              className="btn-group-item"
            >
              <button
                className="btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
            </div>
            <div
              className="btn-group-item"
            >
              <button
                className="btn btn-secondary"
                id="remove_sla_button"
                onClick={[Function]}
                type="button"
              >
                ok
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
